src/inspector/inspector_socket_server.cpp

```diff
@@ -338,10 +338,17 @@ bool InspectorSocketServer::HasTargets() {
 
 bool InspectorSocketServer::Start() {
   state_ = ServerState::kRunning;
-  std::unique_lock<std::mutex> tcp_server_lock(mutex_tcp_server_);
-  std::thread([this, lock = std::move(tcp_server_lock)]() {
-    tcp_server_ = std::make_shared<tcp_server>(
-        port_, InspectorSocketServer::SocketConnectedCallback, this);
+
+  std::thread([this]() {

```

Comment:
>this [](http://example.com/codeflow?start=15&length=4)

Same as the other comment, what ensures that object stays valid by the time that thread executes?